### ISSUE TITLE
lint for test #[ignore] without message

### ISSUE THREAD

**Issue by JustusFluegel** [rust-lang/rust-clippy#13066]
### What it does

Warn on ignored tests without messages

### Advantage

- Tool for projects to make sure that all ignored tests have a valid reason to be ignored

### Drawbacks

- This definitely should not be enabled by default since it's a pretty opinionated lint.

### Example

```rust
#[test]
#[ignore]
fn test() { }
```

Could be written as:

```rust
#[test]
#[ignore = "Some good reason"]
fn test() { }
```

**Comment by blyxyas** [comment]
`style` lints are warn by default. Maybe `pedantic`?

**Comment by JustusFluegel** [comment]
> `style` lints are warn by default. Maybe `pedantic`?

I would put it in the same category as `clippy::allow_attributes_without_reason`, so probably `restriction`
